### PR TITLE
Chore add headers that improve security in nginx

### PR DESCRIPTION
**Pull request by bartoszbetka**
#507

### CODE REVIEW

File: sheetstorm-deployment/consts.yml [hunk 1]
```diff
@@ -10,3 +10,19 @@ django_fixtures_dir:            "{{ sheetstorm_dir }}/sheetstorm/fixtures"
...
 postgres_backup_public_key_dir: "{{ home_dir }}/.key"
 postgres_backup_dir:            "{{ home_dir }}/postgresql_backup"
+
+# Nginx configuration of Content Security Policy header
+content_security_policy_header:
+  allow_script_sources:
+    - https://maxcdn.bootstrapcdn.com
+    - https://cdnjs.cloudflare.com
+    - https://code.jquery.com
+    - https://ajax.googleapis.com
```
Comment (kbeker) [inline sheetstorm-deployment/consts.yml]: Please change to alphabetical order

File: sheetstorm-deployment/consts.yml [hunk 2]
```diff
@@ -10,3 +10,19 @@ django_fixtures_dir:            "{{ sheetstorm_dir }}/sheetstorm/fixtures"
...
+  allow_stylesheet_sources:
+    - https://maxcdn.bootstrapcdn.com
+    - https://code.jquery.com
+    - https://cdnjs.cloudflare.com
```
Comment (kbeker) [inline sheetstorm-deployment/consts.yml]: Same here

File: sheetstorm-deployment/consts.yml [hunk 3]
```diff
@@ -10,3 +10,19 @@ django_fixtures_dir:            "{{ sheetstorm_dir }}/sheetstorm/fixtures"
...
+  allow_font_sources:
+    - https://fonts.googleapis.com
+    - https://fonts.gstatic.com
+    - https://maxcdn.bootstrapcdn.com
```
Comment (kbeker) [inline sheetstorm-deployment/consts.yml]: Lucky good alphabetical order :P

File: sheetstorm-deployment/roles/configure_nginx/templates/maintenance-remote.j2 [hunk 4]
```diff
@@ -4,6 +4,15 @@ limit_conn_zone $binary_remote_addr zone=ip_connection_limit:10m;
 # Disable sending the nginx version in error pages and server header
 server_tokens off;
 
+# Don't allow the browser to render the page inside an frame
+add_header X-Frame-Options SAMEORIGIN;
+
+# Add X-Content-Type-Options header that disables concent-type sniffing on some browsers
+add_header X-Content-Type-Options nosniff;
```
Comment (kbeker) [inline sheetstorm-deployment/roles/configure_nginx/templates/maintenance-remote.j2]: I added this also to django settings. We can test it later if it is needed in both places.

File: sheetstorm-deployment/roles/configure_nginx/templates/sheetstorm-remote.j2 [hunk 5]
```diff
@@ -47,6 +56,11 @@ server {
     include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
     ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
 
+    # Enable HSTS header
+    add_header Strict-Transport-Security "max-age=31536000; includeSubdomains; preload";
+
+    # Add Content Security Policy header that defines from which sources, browser can download content.
+    add_header Content-Security-Policy "base-uri 'self'; form-action 'self'; default-src 'none'; script-src 'self' {% for source in content_security_policy_header.allow_script_sources %}{{ source }} {% endfor %}; connect-src 'self'; img-src 'self'; font-src 'self' {% for source in content_security_policy_header.allow_font_sources %}{{ source }} {% endfor %}; style-src 'self' {% for source in content_security_policy_header.allow_stylesheet_sources %}{{ source }} {% endfor %}; object-src 'none'; frame-ancestors 'none';" always;
```
Comment (kbeker) [inline sheetstorm-deployment/roles/configure_nginx/templates/sheetstorm-remote.j2]: You add to `sheetstorm-deployment/roles/configure_nginx/templates/sheetstorm-remote.j2` and `sheetstorm-deployment/roles/configure_nginx/templates/maintenance-remote.j2` same things. I'm not an expert in deployment things but can't you extract it to somewhere where these two files can import from?

Comment (bartoszbetka) [reply sheetstorm-deployment/roles/configure_nginx/templates/sheetstorm-remote.j2]: This should be consider as separate pull request with refactor.

**Pull request merged**